pkg/api/utils/apiServiceUtils.go

```diff
@@ -27,29 +27,40 @@ func post(uri string, data []byte, api APIService) (*models.Error, error) {
 
 	resp, err := api.getHTTPClient().Do(req)
 	if err != nil {
-		return nil, err
+		return nil, buildErrorResponse(err.Error())
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode >= 200 && resp.StatusCode < 300 {
-		return nil, nil
+	if resp.StatusCode == 200 {
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, buildErrorResponse(err.Error())
+		}
+
+		var channelInfo models.ChannelInfo
+		err = json.Unmarshal(body, &channelInfo)
+		if err != nil {
+			return nil, buildErrorResponse(err.Error())
+		}
+
+		return &channelInfo, nil
 	}
 
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return nil, err
+		return nil, buildErrorResponse(err.Error())
 	}
 
 	var respErr models.Error
 	err = json.Unmarshal(body, &respErr)
 	if err != nil {
-		return nil, err
+		return nil, buildErrorResponse(err.Error())
 	}
 
-	return &respErr, nil
+	return nil, &respErr
 }
 
-func delete(uri string, api APIService) (*models.Error, error) {
+func delete(uri string, api APIService) (*models.ChannelInfo, *models.Error) {
```

Comment:
This looks like as if it is API breaking. Will this cause any issues with the remaining project?

Reply:
No, this is no breaking change since it is aligned with the Keptn CLI, which is the only component using this new utils package.